AsyncDisplayKit/ASMapNode.h

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+#import <MapKit/MapKit.h>
+@interface ASMapNode : ASControlNode
+- (instancetype)initWithCoordinate:(CLLocationCoordinate2D)coordinate NS_DESIGNATED_INITIALIZER;
+
+@property (nonatomic, readonly) ASImageNode *mapImage;
+@property (nonatomic, readonly) ASDisplayNode *liveMap;
+/**
+ Whether the map snapshot should turn into a MKMapView when tapped on. Defaults to YES.
+ */
+@property (nonatomic, assign) BOOL hasLiveMap;
```

Comment:
@aaronschubert0 ok, sounds good, let's remove the default touch handling.  Leave it as an ASControlNode subclass so it's easy for a developer to implement.  Let's actually call it "liveMap / getter = isLiveMap" - then you could do something like -setLiveMap:YES or check mapNode.isLiveMap.

The alternative would be a method like -becomeLiveMap, but then we need one for -removeLiveMap too.  The benefit there would be you could create a direct target / action pair where the action literally is just -becomeLiveMap.  However, I think it will be necessary to change the target / action afterwards anyway or another tap on the live map would disable it, which is probably not desired, as a user might just be trying to double tap to zoom in or something.  So requiring a developer to create their own action method (in which they'll set _mapNode.liveMap = YES) makes sense.  I'm open to different name ideas for sure!